Helpers.gs

```diff
@@ -134,7 +134,32 @@ function fetchSourceCalendars(sourceCalendarURLs){
     var colorId = source[1];
     
     callWithBackoff(function() {
-      var urlResponse = UrlFetchApp.fetch(url, { 'validateHttpsCertificates' : false, 'muteHttpExceptions' : true });
+      var urlReponse;
+      // Follow https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1 taking into account that:
+      // - UrlFetchApp.fetch only accepts https and http links
+      // it defaults to http if no scheme is given, but just // is not accepted
+      // - Google Apps Scripts does not seem to accept \] in [^] lists
+      // - to have the less possible need of percent-encoding, as much characters as possible are accepted
+      const urlRegex = RegExp("^(https?://)?(?<username>[^:/@]+)(?::(?<password>[^/@]*))?@");
```

Comment:
Thanks for converting this here based on feedback. I have one final suggestion: let's break this code block out to a separate function (eg `parseAuthenticationFromCalendarURL` or whatever)

Reply:
You're welcome.
If you would like it in a separate function, please go ahead. You can take the code and put it in the structure you want, no problem.